db-engine/src/main/java/com/splicemachine/db/impl/ast/CorrelatedPushDown.java

```diff
@@ -26,15 +26,15 @@
 package com.splicemachine.db.impl.ast;
 
 
-import com.google.common.base.Function;
+import org.spark_project.guava.base.Function;
 import com.splicemachine.db.iapi.error.StandardException;
 import com.splicemachine.db.iapi.sql.compile.Optimizable;
 import com.splicemachine.db.iapi.sql.compile.Visitable;
 import com.splicemachine.db.impl.sql.compile.*;
 import org.apache.commons.lang3.tuple.Pair;
 import org.apache.log4j.Logger;
-import org.sparkproject.guava.collect.Collections2;
-import org.sparkproject.guava.collect.Iterables;
```

Comment:
I think it would be a better idea to use our own Guava (possibly shading it ourselves under com.splicemachine.guava) rather than depending on Spark's shaded guava:
- They could change their guava version or we might want to upgrade.
- Their use of guava is entirely internal, they don't expose Guava on their public API, and they could change it at any time.

What do you think?